server/backend/api/app/crud/crud_application.py

```diff
@@ -108,53 +106,76 @@ def get_application_roles(
+    :param requester: the user who perform this request/action.
+    :return: the user role assignments for the given application.
     """
-    LOGGER.debug(f"Query for user role assignments for app id: {application_id}")
+    LOGGER.debug(f"Querying for user role assignments on app id:\
+                  {application_id} by requester: {requester} ")
 
-    crossref = (
+    # base query - users assigned to the application. This could be the case
+    #              for [APP]_ADMIN.
+    q = (
         db.query(models.FamUserRoleXref)
         .join(models.FamRole)
         .filter(models.FamRole.application_id == application_id)
-        .all()
     )
 
-    LOGGER.debug(f"Query for user role assignment complete with # of results = {len(crossref)}")
-    return crossref
-
-
-def get_application_id_by_role_id(db: Session, role_id):
-    # question: why we need to go to fam_application table?
-    # the fam_role table has the application_id already
-
-    application_id =\
-        db.query(models.FamApplication.application_id)\
-        .join(models.FamRole)\
-        .filter(models.FamRole.role_id == role_id)\
-        .scalar()
-
-    return application_id
-
+    if not crud_utils.is_app_admin(
+        application_id=application_id,
+        access_roles=requester.access_roles,
+        db=db
+    ):
+        # subquery for finding out what roles (role_ids) the requester
+        # (as an application delegated admin) is managing at.
+        role_ids_dlgdadmin_managed_subquery = (
+            db.query(models.FamAccessControlPrivilege.role_id)
+            .join(models.FamUser)
+            .join(models.FamRole)
+            .filter(
+                models.FamUser.cognito_user_id == requester.cognito_user_id,
+                models.FamRole.application_id == application_id
+            )
+            .subquery()
+        )
 
-def get_application_id_by_user_role_xref_id(db: Session, user_role_xref_id):
-    application_id =\
-        db.query(models.FamApplication.application_id)\
-        .join(models.FamRole)\
-        .join(models.FamUserRoleXref)\
-        .filter(models.FamUserRoleXref.user_role_xref_id == user_role_xref_id)\
-        .scalar()
+        # filtered by the managed role for user_role assignments that the
+        # requester (as an delegated admin) is allowed to see.
+        q = q.filter(
+            models.FamUserRoleXref.role_id.in_(
+                select(role_ids_dlgdadmin_managed_subquery)
+            )
+        )
 
-    return application_id
+        if (requester.user_type_code == UserType.BCEID):
+            # append additional filtering: A BCeID requester can only see
+            # user_role records belonging to the same business organization.
+
+            # Note, need to reassign to the variable from the base query.
+            q = (
+                q
+                .join(models.FamUser)
+                # .join(models.FamUserRoleXref)
```

Comment:
if this is not in used anymore maybe we can remove it?

Reply:
Oh, boy.... I missed this thing again?.....too bad....no more programming.... 
Thanks Catherine for noticing this!!